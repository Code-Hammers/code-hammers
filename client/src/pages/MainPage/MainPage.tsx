import React from "react";
import { useAppSelector } from "../../app/hooks";

const MainPage: React.FC = () => {
  const user = useAppSelector((state) => state.user.userName);

  return (
    <div className="min-h-screen bg-gray-100 flex flex-col items-center justify-center">
      <h1 className="text-4xl font-extrabold mb-4">{user}</h1>
      <h1>TEST</h1>
    </div>
  );
};

export default MainPage;
