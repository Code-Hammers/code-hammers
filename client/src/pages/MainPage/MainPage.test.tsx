import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import MainPage from './MainPage';

interface State {
  user: {
    userName: string;
  };
}

const mockStore = configureStore<State>([]);
const initialState: State = {
  user: {
    userName: 'TEST',
  },
};

describe('MainPage Component', () => {
  it('renders correctly', () => {
    const store = mockStore(initialState);
    const tree = create(
      <Provider store={store}>
        <MainPage />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
