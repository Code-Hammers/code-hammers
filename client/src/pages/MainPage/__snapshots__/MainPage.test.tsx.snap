// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders correctly 1`] = `
<div
  className="pt-40 min-h-screen bg-gray-900 text-white flex flex-col items-center justify-center p-4"
>
  <h1
    className="text-4xl font-extrabold mb-4"
  >
    Welcome!
  </h1>
  <section
    className="w-full max-w-4xl p-4 mt-4 bg-gradient-to-l from-gray-700 via-gray-800 to-gray-900 rounded-lg shadow-lg flex md:flex-row flex-col items-center gap-4"
  >
    <img
      alt="Description of Image 1"
      className="flex-shrink-0 w-full md:w-1/2 rounded-lg object-cover"
      src="https://picsum.photos/200/300?grayscale"
      style={
        {
          "maxHeight": "300px",
        }
      }
    />
    <div
      className="flex-grow"
    >
      <h3
        className="text-2xl font-bold mb-2"
      >
        About Code Hammers
      </h3>
      <p
        className="text-gray-300"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum in ligula eu erat sollicitudin porttitor. Nunc congue mi eget nunc dapibus dictum.Duis tempus diam ac massa bibendum, sed tempus sapien posuere. In elementum enim ac dui interdum porta. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac.
      </p>
    </div>
  </section>
  <section
    className="w-full max-w-4xl p-4 mt-4 bg-gradient-to-r from-gray-700 via-gray-800 to-gray-900 rounded-lg shadow-lg flex md:flex-row flex-col items-center gap-4"
  >
    <img
      alt="Description of Image 1"
      className="flex-shrink-0 w-full md:w-1/2 rounded-lg object-cover md:order-last"
      src="https://picsum.photos/200/300?grayscale"
      style={
        {
          "maxHeight": "300px",
        }
      }
    />
    <div
      className="flex-grow"
    >
      <h3
        className="text-2xl font-bold mb-2"
      >
        Features
      </h3>
      <p
        className="text-gray-300"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum in ligula eu erat sollicitudin porttitor. Nunc congue mi eget nunc dapibus dictum. Duis tempus diam ac massa bibendum, sed tempus sapien posuere. In elementum enim ac dui interdum porta. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac.
      </p>
    </div>
  </section>
  <section
    className="w-full max-w-4xl p-4 mt-4 bg-gradient-to-l from-gray-700 via-gray-800 to-gray-900 rounded-lg shadow-lg flex md:flex-row flex-col items-center gap-4"
  >
    <img
      alt="Description of Image 1"
      className="flex-shrink-0 w-full md:w-1/2 rounded-lg object-cover"
      src="https://picsum.photos/200/300?grayscale"
      style={
        {
          "maxHeight": "300px",
        }
      }
    />
    <div
      className="flex-grow"
    >
      <h3
        className="text-2xl font-bold mb-2"
      >
        Get Started!
      </h3>
      <p
        className="text-gray-300"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum in ligula eu erat sollicitudin porttitor. Nunc congue mi eget nunc dapibus dictum.Duis tempus diam ac massa bibendum, sed tempus sapien posuere. In elementum enim ac dui interdum porta. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac.
      </p>
    </div>
  </section>
  <section
    className="w-full max-w-4xl p-4 mt-4 bg-gradient-to-r from-gray-700 via-gray-800 to-gray-900 rounded-lg shadow-lg flex md:flex-row flex-col items-center gap-4"
  >
    <img
      alt="Description of Image 1"
      className="flex-shrink-0 w-full md:w-1/2 rounded-lg object-cover md:order-last"
      src="https://picsum.photos/200/300?grayscale"
      style={
        {
          "maxHeight": "300px",
        }
      }
    />
    <div
      className="flex-grow"
    >
      <h3
        className="text-2xl font-bold mb-2"
      >
        Boopity Bops
      </h3>
      <p
        className="text-gray-300"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum in ligula eu erat sollicitudin porttitor. Nunc congue mi eget nunc dapibus dictum. Duis tempus diam ac massa bibendum, sed tempus sapien posuere. In elementum enim ac dui interdum porta. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac.
      </p>
    </div>
  </section>
  <section
    className="w-full max-w-4xl p-4 mt-4 bg-gradient-to-l from-gray-700 via-gray-800 to-gray-900 rounded-lg shadow-lg flex md:flex-row flex-col items-center gap-4"
  >
    <img
      alt="Description of Image 1"
      className="flex-shrink-0 w-full md:w-1/2 rounded-lg object-cover"
      src="https://picsum.photos/200/300?grayscale"
      style={
        {
          "maxHeight": "300px",
        }
      }
    />
    <div
      className="flex-grow"
    >
      <h3
        className="text-2xl font-bold mb-2"
      >
        Bopitty Boops
      </h3>
      <p
        className="text-gray-300"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum in ligula eu erat sollicitudin porttitor. Nunc congue mi eget nunc dapibus dictum.Duis tempus diam ac massa bibendum, sed tempus sapien posuere. In elementum enim ac dui interdum porta. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac.
      </p>
    </div>
  </section>
</div>
`;
